### PR TITLE
Bug Fixes for Typechecking in Tally Aggregation

### PR DESCRIPTION
**Pull request by wbinventor**
This PR fixes a few bugs that I came across when performing tally aggregation (*e.g.*, `Tally.summation(...)`). In particular, when performing tally aggregation across derived tallies, such as those created by `openmc.mgxs` failed due to some typechecking issues in `openmc.aggregation`. This was resolved by expanding the accepted types for those checks, and moving `openmc.aggregate` into `openmc.cross` to resolve circular dependencies between those two modules.

### CODE REVIEW

**Comment by wbinventor** [comment]
I added a single line commit to this PR which fixes a bug in the `StatePoint.with_summary` property. This is an unrelated bug fix, but thought I'd tack it on here to avoid a further explosion of PRs to the repo.

File: openmc/statepoint.py [hunk 1]
```diff
@@ -451,7 +451,7 @@ def summary(self):
 
     @property
     def with_summary(self):
-        return False if self.summary is None else True
+        return False if self.summary is False else True
```
Comment (paulromano) [inline openmc/statepoint.py]: This property doesn't seem terribly useful. Is there a reason to keep it around or should we just get rid of it? I don't see it used anywhere actually.

Comment (wbinventor) [inline openmc/statepoint.py]: It is used a few times for error checking in `openmc.mgxs`. I use it in a number of places in InferMC which is built atop of `openmc.mgxs`. I can see removing it in place of the `summary` property though.

Comment (wbinventor) [inline openmc/statepoint.py]: I just removed it since we can live with `StatePoint.summary` in place of both properties.

**Comment by paulromano** [comment]
@wbinventor What do you think about renaming `openmc.cross` to `openmc.arithmetic` or something else? I feel like "cross" isn't broad enough now.

**Comment by wbinventor** [comment]
I renamed "cross.py" to "arithmetic.py." I think sub-module'ing the tally-related code can wait for another PR.

**Pull request merged**